File: VB/MapControl_PrintingExporting/Form1.vb
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraMap
Imports DevExpress.XtraPrinting
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Drawing.Imaging
Imports System.Windows.Forms

Namespace MapControl_PrintingExporting
    Public Enum PrintType
        Preview
        RibbonPreview
        Dialog
    End Enum

    Partial Public Class Form1
        Inherits Form

        Private Const printButtonTextBase As String = "Show "
        Private Const exportButtonTextBase As String = "Export to "
        Private Const filepathBase As String = "Exported Map"

        Private printType As PrintType = MapControl_PrintingExporting.PrintType.Preview
        Private exportFormat As ExportFormat = ExportFormat.Image

        Private printHashtable As New Hashtable()
        Private exportHashtable As New Hashtable()

        Public Sub New()
            InitializeComponent()
            InitializePrintButton()
            InitializeExportButton()
            InitializeSizeModeComboBox()
        End Sub

        #Region "#InitSizeModes"
        Private Sub InitializeSizeModeComboBox()
            cbSizeMode.DataSource = System.Enum.GetValues(GetType(MapPrintSizeMode))
        End Sub
        #End Region ' #InitSizeModes

        Private Sub InitializePrintButton()
            Dim popup As New DXPopupMenu()
            Dim item As New DXMenuItem("Print Preview", AddressOf printPopupItem_Click)
            printHashtable.Add(item, MapControl_PrintingExporting.PrintType.Preview)
            popup.Items.Add(item)

            item = New DXMenuItem("Ribbon Print Preview", AddressOf printPopupItem_Click)
            printHashtable.Add(item, MapControl_PrintingExporting.PrintType.RibbonPreview)
            popup.Items.Add(item)

            item = New DXMenuItem("Print Dialog", AddressOf printPopupItem_Click)
            printHashtable.Add(item, MapControl_PrintingExporting.PrintType.Dialog)
            popup.Items.Add(item)

            ddbPrint.DropDownControl = popup
            ddbPrint.Text = printButtonTextBase & printType.ToString()
        End Sub

        Private Sub InitializeExportButton()
            Dim popup As New DXPopupMenu()

            Dim item As New DXMenuItem(ExportFormat.Image.ToString(), AddressOf exportPopupItem_Click)
            exportHashtable.Add(item, ExportFormat.Image)
            popup.Items.Add(item)

            item = New DXMenuItem(ExportFormat.Mht.ToString(), AddressOf exportPopupItem_Click)
            exportHashtable.Add(item, ExportFormat.Mht)
            popup.Items.Add(item)

            item = New DXMenuItem(ExportFormat.Pdf.ToString(), AddressOf exportPopupItem_Click)
            exportHashtable.Add(item, ExportFormat.Pdf)
            popup.Items.Add(item)

            item = New DXMenuItem(ExportFormat.Xls.ToString(), AddressOf exportPopupItem_Click)
            exportHashtable.Add(item, ExportFormat.Xls)
            popup.Items.Add(item)

            item = New DXMenuItem(ExportFormat.Xlsx.ToString(), AddressOf exportPopupItem_Click)
            exportHashtable.Add(item, ExportFormat.Xlsx)
            popup.Items.Add(item)

            ddbExport.DropDownControl = popup
            ddbExport.Text = exportButtonTextBase & exportFormat.ToString()
        End Sub

        Private Sub printPopupItem_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim item As DXMenuItem = TryCast(sender, DXMenuItem)
            ddbPrint.Text = printButtonTextBase & item.Caption
            printType = DirectCast(printHashtable(item), PrintType)
        End Sub

        Private Sub exportPopupItem_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim item As DXMenuItem = TryCast(sender, DXMenuItem)
            ddbExport.Text = exportButtonTextBase & item.Caption
            exportFormat = DirectCast(exportHashtable(item), ExportFormat)
        End Sub

        #Region "#SizeModeChanged"
        Private Sub cbSizeMode_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbSizeMode.SelectedIndexChanged
            Dim mode As MapPrintSizeMode = DirectCast(cbSizeMode.SelectedValue, MapPrintSizeMode)
            mapControl.PrintOptions.SizeMode = mode
        End Sub
        #End Region ' #SizeModeChanged

        #Region "#Printing"
        Private Sub ddbPrint_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ddbPrint.Click
            Select Case printType
                Case (MapControl_PrintingExporting.PrintType.Preview)
                    mapControl.ShowPrintPreview()
                Case (MapControl_PrintingExporting.PrintType.RibbonPreview)
                    mapControl.ShowRibbonPrintPreview()
                Case (MapControl_PrintingExporting.PrintType.Dialog)
                    mapControl.ShowPrintDialog()
            End Select
        End Sub
        #End Region ' #Printing

        #Region "#Exporting"
        Private Sub ddbExport_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ddbExport.Click
            Dim filepath As String = filepathBase & exportFormat.ToString()
            Select Case exportFormat
                Case (ExportFormat.Image)
                    mapControl.ExportToImage(filepath, ImageFormat.Jpeg)
                Case (ExportFormat.Mht)
                    mapControl.ExportToMht(filepathBase)
                Case (ExportFormat.Pdf)
                    mapControl.ExportToPdf(filepathBase)
                Case (ExportFormat.Xls)
                    mapControl.ExportToXls(filepathBase)
                Case (ExportFormat.Xlsx)
                    mapControl.ExportToXlsx(filepathBase)
                Case Else
                    MessageBox.Show("Export to a *" & exportFormat.ToString() & " file format is impossible.", "Export to " & exportFormat.ToString())
                    Return
            End Select

            MessageBox.Show("Export to a *" & exportFormat.ToString() & " file is done.", "Export to " & exportFormat.ToString())
        End Sub
        #End Region ' #Exporting
    End Class
End Namespace

File: CS/MapControl_PrintingExporting/Form1.cs
using DevExpress.Utils.Menu;
using DevExpress.XtraMap;
using DevExpress.XtraPrinting;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing.Imaging;
using System.Windows.Forms;

namespace MapControl_PrintingExporting {
    public enum PrintType { Preview, RibbonPreview, Dialog };
    
    public partial class Form1 : Form {
        const string printButtonTextBase = "Show ";
        const string exportButtonTextBase = "Export to ";
        const string filepathBase = "Exported Map";

        PrintType printType = PrintType.Preview;
        ExportFormat exportFormat = ExportFormat.Image;
        
        Hashtable printHashtable = new Hashtable();
        Hashtable exportHashtable = new Hashtable();

        public Form1() {
            InitializeComponent();
            InitializePrintButton();
            InitializeExportButton();
            InitializeSizeModeComboBox();
        }

        #region #InitSizeModes
        void InitializeSizeModeComboBox() {
            cbSizeMode.DataSource = Enum.GetValues(typeof(MapPrintSizeMode));
        }
        #endregion #InitSizeModes
        
        void InitializePrintButton() {
            DXPopupMenu popup = new DXPopupMenu();
            DXMenuItem item = new DXMenuItem("Print Preview", printPopupItem_Click);
            printHashtable.Add(item, PrintType.Preview);
            popup.Items.Add(item);

            item = new DXMenuItem("Ribbon Print Preview", printPopupItem_Click);
            printHashtable.Add(item, PrintType.RibbonPreview);
            popup.Items.Add(item);

            item = new DXMenuItem("Print Dialog", printPopupItem_Click);
            printHashtable.Add(item, PrintType.Dialog);
            popup.Items.Add(item);

            ddbPrint.DropDownControl = popup;
            ddbPrint.Text = printButtonTextBase + printType.ToString();           
        }

        void InitializeExportButton() {
            DXPopupMenu popup = new DXPopupMenu();           

            DXMenuItem item = new DXMenuItem(ExportFormat.Image.ToString(), exportPopupItem_Click);
            exportHashtable.Add(item, ExportFormat.Image);
            popup.Items.Add(item);

            item = new DXMenuItem(ExportFormat.Mht.ToString(), exportPopupItem_Click);
            exportHashtable.Add(item, ExportFormat.Mht);
            popup.Items.Add(item);

            item = new DXMenuItem(ExportFormat.Pdf.ToString(), exportPopupItem_Click);
            exportHashtable.Add(item, ExportFormat.Pdf);
            popup.Items.Add(item);

            item = new DXMenuItem(ExportFormat.Xls.ToString(), exportPopupItem_Click);
            exportHashtable.Add(item, ExportFormat.Xls);
            popup.Items.Add(item);

            item = new DXMenuItem(ExportFormat.Xlsx.ToString(), exportPopupItem_Click);
            exportHashtable.Add(item, ExportFormat.Xlsx);
            popup.Items.Add(item);

            ddbExport.DropDownControl = popup;
            ddbExport.Text = exportButtonTextBase + exportFormat.ToString();
        }

        void printPopupItem_Click(object sender, EventArgs e) {
            DXMenuItem item = sender as DXMenuItem;
            ddbPrint.Text = printButtonTextBase + item.Caption;
            printType = (PrintType)printHashtable[item];
        }

        void exportPopupItem_Click(object sender, EventArgs e) {
            DXMenuItem item = sender as DXMenuItem;
            ddbExport.Text = exportButtonTextBase + item.Caption;
            exportFormat = (ExportFormat)exportHashtable[item];
        }

        #region #SizeModeChanged
        private void cbSizeMode_SelectedIndexChanged(object sender, EventArgs e) {
            MapPrintSizeMode mode = (MapPrintSizeMode)cbSizeMode.SelectedValue;
            mapControl.PrintOptions.SizeMode = mode;
        }
        #endregion #SizeModeChanged

        #region #Printing
        private void ddbPrint_Click(object sender, EventArgs e) {
            switch (printType) {
                case (PrintType.Preview):
                    mapControl.ShowPrintPreview();
                    break;
                case (PrintType.RibbonPreview):
                    mapControl.ShowRibbonPrintPreview();
                    break;
                case (PrintType.Dialog):
                    mapControl.ShowPrintDialog();
                    break;
            }
        }
        #endregion #Printing

        #region #Exporting
        private void ddbExport_Click(object sender, EventArgs e) {
            string filepath = filepathBase + exportFormat.ToString();
            switch (exportFormat) {
                case (ExportFormat.Image):
                    mapControl.ExportToImage(filepath, ImageFormat.Jpeg);
                    break;
                case (ExportFormat.Mht):
                    mapControl.ExportToMht(filepathBase);
                    break;
                case (ExportFormat.Pdf):
                    mapControl.ExportToPdf(filepathBase);
                    break;
                case (ExportFormat.Xls):
                    mapControl.ExportToXls(filepathBase);
                    break;
                case (ExportFormat.Xlsx):
                    mapControl.ExportToXlsx(filepathBase);
                    break;
                default:
                    MessageBox.Show("Export to a *" + exportFormat.ToString() + " file format is impossible.", "Export to " + exportFormat.ToString());
                    return;
            }

            MessageBox.Show("Export to a *" + exportFormat.ToString() + " file is done.", "Export to " + exportFormat.ToString());
        }
        #endregion #Exporting
    }
}
